packages/nimble/inst/CppCode/nimOptim.cpp

```diff
@@ -0,0 +1,51 @@
+#include <R_ext/Applic.h>
+#include <algorithm>
+#include <cmath>
+#include <limits>
+#include <nimble/nimOptim.h>
+
+// This wraps a NimObjectiveFn as an R `optimfn` type
+//   typedef double optimfn(int, double *, void *);
+// defined in
+//   https://svn.r-project.org/R/trunk/src/include/R_ext/Applic.h
+static double optimfn_wrapper(int n, double* par, void* fn) {
+    NimArr<1, double> nim_par;
+    nim_par.setSize(n);
```

Comment:
It is slightly more efficient to use `nim_par.setSize(n, false, false)`, which makes no initialization be done.  I would have never thought it would be noticeable, we saw in MCMCs that unnecessary initialization (at the time we used std::vector as the underlying memory allocation, which always initializes) took up a meaningful amount of computation time when repeats zillions of times.